packages/mobile/src/navigator/NavigationService.ts

```diff
@@ -19,14 +19,15 @@ const TAG = 'NavigationService'
 type SafeNavigate = typeof navigate
 
 export const navigationRef = createRef<NavigationContainerRef>()
+export const isReadyRef: MutableRefObject<boolean | null> = createRef()
```

Comment:
Can we also initialize in `false` so we don't have the `| null` on the type? It would also avoid the `useEffect` on the `NavigationWrapper`

Reply:
I am not very familiar with the nuances of these functions but createRef doesn't seem to take an initial value. Should I be using something like useState?

Also setting the isReady ref to false is sometimes good when the component gets unmounted for example. It probably doesn't happen often that it gets unmounted but then mounted back but maybe on android with the back button it does so just to be safe let's keep it?

Reply:
Thanks to Gonza I now understand better how these effects work.

createRef does not take an initial value unfortunately so we just decided to init to false right under.

We also agreed to keep the effect separate since it looks cleaner and also we need it because it returns a function to be used on unmount.